python/delta_sharing/rest_client.py

```diff
@@ -71,6 +73,13 @@ class QueryTableMetadataResponse:
     metadata: Metadata
 
 
+@dataclass(frozen=True)
+class QueryTableDeltaMetadataResponse:
```

Comment:
I feel we could just add fields into existing `QueryTableMetadataResponse`, and add one more string field indicating the responded format.

Reply:
I was thinking about this but logically I felt the response should only contain what is needed for the response type. If it is not using delta, why have empty fields for it when we can create a new object to handle it. I am fine either way though